portfolio/src/main/webapp/script.js

```diff
@@ -175,18 +222,314 @@ async function deleteAllComments() {
+
+        // Populate form with name options
+        let nameSelect = document.getElementById('comment_name');
+
+        const emailOption = document.createElement('option');
+        emailOption.innerText = user.email;
+        emailOption.setAttribute('value', user.email);
+
+        const nicknameOption = document.createElement('option');
+        nicknameOption.innerText = user.nickname;
+        nicknameOption.setAttribute('value', user.nickname);
+
+        nameSelect.appendChild(nicknameOption);
+        nameSelect.appendChild(emailOption);
+
+        getComments();
+    } else {
+        commentSection.style.display = 'none';
+        commentBlocker.style.display = 'block';
+
+        // Make log in button 
+        const loginButton = document.createElement('a')
+        loginButton.setAttribute('href', user.loginURL);
+        loginButton.innerText = 'Log in';
+        loginButton.classList.add('link');
 
+        // Attach to parent nodes
+        commentBlocker.appendChild(loginButton);
+    }
 }
 
+// Location data for all content
+const markerData =  [new google.maps.LatLng(37.422179, -122.084036), 
+    new google.maps.LatLng(33.809218, -118.066596),
+    new google.maps.LatLng(37.428214, -122.161195),
+    new google.maps.LatLng(37.425333, -122.170153),
+    new google.maps.LatLng(37.430174, -122.173341),
+    new google.maps.LatLng(42.383904, -71.133425),
+    new google.maps.LatLng(19.419853, -99.161854),
+    new google.maps.LatLng(37.427474, -122.169901),
+    new google.maps.LatLng(30.302448, -97.748066),
+    new google.maps.LatLng(44.061074, -70.532018),
+    new google.maps.LatLng(42.360162, -71.094203),
+    new google.maps.LatLng(39.952391, -75.193171)]
+
 /**
- * Init function to perform certain tasks onload
+ * Generates the map & markers
+ */
+function createMap() {
+    let styledMapType = new google.maps.StyledMapType(
+        mapJSON,
+        {name: 'My Map'});
+
+    const map = new google.maps.Map(
+        document.getElementById('map'),
+        {center: {lat: 37.693622, lng: -97.195151}, zoom: 3,
+        mapTypeControlOptions: {
+                mapTypeIds: ['My Map']
+            }
+        });
+  
+    map.mapTypes.set('My Map', styledMapType);
+    map.setMapTypeId('My Map');
+
+    // Gather HTML for marker content & generate markers
+    prepareMarkers();
+    for (let i = 0; i < markersHTML.length; i++) {
+        addMarker(map, i);
+    }
+}
+
+/**
+ * Adds content to a marker and adds to map
+ */
+function addMarker(map, i) {
+    // Add marker content
+    let infowindow = new google.maps.InfoWindow({
+        content: markersHTML[i]
+    });
+
+    // Create marker 
+    let marker = new google.maps.Marker({
+        position: markerData[i],
+        map: map,
+        title: 'marker #' + i,
+    });
+
+    // Show information when clicked
+    marker.addListener('click', function() {
+        infowindow.open(map, marker);
+    });
+}
+
+let markersHTML = [];
```

Comment:
so that I don't need to fetch the HTML each time the map tab is opened..... it fetches it the first time then saves it as a global variable

Reply:
maybe put it at the top with the other globals then? usually this would be saved on the user's device (i.e. in a cookie) or in some global state system